pgmpy/prediction/NaiveBackdoorRegressor.py

```diff
@@ -0,0 +1,284 @@
+    """
+    A naive backdoor regressor that uses causal graph roles for feature selection.
+
+    This estimator combines exposure, adjustment, and pretreatment variables as features
+    to predict the outcome variable. The approach is "naive" because it assumes the
+    adjustment set is already correctly identified in the causal graph roles and simply
+    concatenates these variables as features for standard ML prediction.
+
+    Parameters
+    ----------
+    causal_graph : DAG, PDAG, or ADMG
+        A pgmpy causal graph object (DAG, PDAG, or ADMG) with defined roles for
+        exposure, outcome, and adjustment variables. Must have exactly one exposure
+        and one outcome variable. The 'adjustment' role must be explicitly defined
+        even if empty to prevent accidental omission of confounders.
+    base_estimator : sklearn estimator, optional (default=LinearRegression())
+        The base sklearn estimator to use for prediction.
+    multi_output : bool, optional (default=False)
+        Reserved for future multiple outcome support. Currently must be False.
+
+    Attributes
+    ----------
+    `estimator_` : sklearn estimator
+        The fitted base estimator.
+    `feature_names_in_` : ndarray of shape (n_features,)
+        Names of features seen during fit.
+    `n_features_in_` : int
+        Number of features seen during fit.
+    `exposure_var_` : str
+        Name of exposure variable extracted from causal graph.
+    `adjustment_vars_` : list
+        List of adjustment variable names extracted from causal graph.
+    `pretreatment_vars_` : list
+        List of pretreatment variable names extracted from causal graph.
+    `outcome_var_` : str
+        Name of outcome variable extracted from causal graph.
+    `feature_columns_` : list
+        List of feature column names used (exposure + adjustment + pretreatment).
+
+    Examples
+    --------
+    >>> from pgmpy.base import DAG
+    >>> from sklearn.ensemble import RandomForestRegressor
+    >>> import pandas as pd
+    >>>
+    >>> # Create DAG with roles
+    >>> dag = DAG(
+    ...     [("Z", "X"), ("Z", "Y"), ("X", "Y")],
+    ...     roles={"exposure": "X", "outcome": "Y", "adjustment": ["Z"]},
+    ... )
+    >>>
+    >>> # Create dummy data
+    >>> data = pd.DataFrame({"X": [1, 2, 3], "Z": [4, 5, 6], "Y": [7, 8, 9]})
+    >>> X_train, y_train = data[["X", "Z"]], data["Y"]
+    >>>
+    >>> regressor = NaiveBackdoorRegressor(
+    ...     causal_graph=dag, base_estimator=RandomForestRegressor()
+    ... )
+    >>> regressor.fit(X_train, y_train)
+    >>> predictions = regressor.predict(X_train)
+    """
+
+    def __init__(
+        self,
+        causal_graph,
+        base_estimator: Optional[BaseEstimator] = None,
+        multi_output=False,
+    ):
+        self.causal_graph = causal_graph
+        self.base_estimator = base_estimator
+        self.multi_output = multi_output
+
+        if multi_output:
+            raise NotImplementedError(
```

Comment:
The sklearn compatibility check `check_do_not_raise_errors_in_init_or_set_params` is failing because of this. The `__init__`  method shouldn't raise any error.